modules/videoio/src/container_avi.cpp

```diff
@@ -742,8 +734,8 @@ void BitStream::patchInt(int val, size_t pos)
     }
     else
     {
-        long fpos = ftell(m_f);
-        fseek(m_f, (long)pos, SEEK_SET);
+        int32_t fpos = ftell(m_f);
+        fseek(m_f, (int32_t)pos, SEEK_SET);
```

Comment:
[ftell](http://www.cplusplus.com/reference/cstdio/ftell/) / [fseek](http://www.cplusplus.com/reference/cstdio/fseek/) use "long int" type (it is still 32-bit on Windows, even in x86-64 mode - so using of fgetpos/fsetpos may help).

Also we need `CV_Assert(pos < (1<<31));` check.